packages/libs/src/sdk/middleware/addAppInfoMiddleware.ts

```diff
@@ -10,22 +10,31 @@ import {
 } from '../api/generated/default'
 
 let appName: string | undefined
+let apiKey: string | undefined
 
 /**
  * Appends the configured app_name to the query string for tracking API usage
  * @param options the middleware options
  * @param {string} options.appName the name of the app using the SDK
  */
-export const addAppNameMiddleware = ({
+export const addAppInfoMiddleware = ({
+  apiKey: providedApiKey,
   appName: providedAppName,
   services
 }: {
+  apiKey?: string
   appName?: string
   services: any
 }): Middleware => {
+  apiKey = providedApiKey
   appName = providedAppName
   return {
     pre: async (context: RequestContext): Promise<FetchParams> => {
+      // If an api key is not provided, use the authed address (hedgehog / external wallet)
+      if (!apiKey) {
+        apiKey = await services.auth.getAddress()
```

Comment:
This doesn't exist always though, right? Won't this throw for every request in the default auth case?

Reply:
yeah you're right. we weren't going into this flow at all because our client was passing appName

Reply:
Updated to handle this elegantly for now